src/extensions/extra/ImageUrlCodeHints/camera/video.js

```diff
@@ -0,0 +1,67 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true,  regexp: true, indent: 4, maxerr: 50 */
+/*global define, window */
+
+// Camera component to stream a video feed from the user's webcam
+define(function (require, exports, module) {
+    "use strict";
+
+    var vendorURL = window.URL || window.webkitURL;
+
+    function Video(context) {
+        this.context = context;
+        this.streaming = false;
+    }
+
+    // Begine playing the video
+    Video.prototype.play = function(stream) {
+        var self = this;
+        var mainInterface = this.context.interface;
+        // We need to initialize all interfaces first
+        mainInterface.init();
+        var videoInterface = this.interface;
+
+        this.stream = stream;
+        videoInterface.src = vendorURL.createObjectURL(stream);
+        videoInterface.addEventListener("canplay", function() {
+            self.startStream();
+        }, false);
+        videoInterface.addEventListener("pause", function() {
+            self.endStream();
+        });
+
+        mainInterface.enableSnap();
+        videoInterface.play();
```

Comment:
See https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events.  Different browsers deal with these events differently, and Chrome is horrible about how it does things.  Also, you should be prepared for browsers to fire events more than once (i.e, remove your listeners once they fire).